v2/internal/controllers/compute_test.go

```diff
@@ -0,0 +1,188 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package controllers_test
+
+import (
+	"fmt"
+	"github.com/Azure/azure-service-operator/v2/internal/util/to"
+	v1 "k8s.io/api/core/v1"
+
+	. "github.com/onsi/gomega"
+
+	compute2020 "github.com/Azure/azure-service-operator/v2/api/compute/v1api20201201"
+	network "github.com/Azure/azure-service-operator/v2/api/network/v1api20201101"
+	resources "github.com/Azure/azure-service-operator/v2/api/resources/v1api20200601"
+
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+)
+
+func newVirtualMachine20201201(
```

Comment:
Is this only exposing helper methosd now? Why not move this too?

Reply:
Makes sense to have them helper methods in the test files

Reply:
Strong disagree. A helper belongs in a test file if it's only used by that test - having tests cross-wired makes things brittle as changes can be made for one test that inadvertently impact on other tests. These helpers are in separate methods to help convey that they're used from multiple places.

Reply:
I don't mind the helpers being in their own file, but:
1. I thought you can't import functions from `_test` files from other packages, so if these need to be used in the other directory they cannot be, can they? In fact, I think in this PR now these same functions are duplicated in both places (old folder and new folder), which feels bad?
2. The name of the file they're in now suggests to me that it contains tests, when it actually doesn't. TBH I am not sure that there is a good way to solve this other than pulling them out into their own package `testcommon` or equivalent. I'm not sure that's worth doing so willing to live with this.

Item 1 above (especially the fact that these are now duplicated) is awkward - again suggesting that we should move the whole folder not just part of it?